core/primitives/src/congestion_info.rs

```diff
@@ -95,6 +93,12 @@ impl CongestionControl {
         }
     }
 
+    pub fn is_fully_congested(congestion_level: f64) -> bool {
+        // note: using float equality is okay here because
+        // `clamped_f64_fraction` clamps to exactly 1.0.
+        congestion_level == 1.0
```

Comment:
unrelated but I'd be tempted to add a debug_assert congestion level <= 1.0

Reply:
I can add it

Reply:
Added